tsl/src/bgw_policy/continuous_aggregate_api.c

```diff
@@ -165,6 +165,49 @@ check_valid_interval(Oid dim_type, Oid interval_type, const char *str_msg)
 	}
 }
 
+static void
+check_valid_interval_values(Oid interval_type, Datum start_interval, Datum end_interval)
+{
+	bool valid = true;
+	if (IS_INTEGER_TYPE(interval_type))
+	{
+		switch (interval_type)
+		{
+			case INT2OID:
+			{
+				if (DatumGetInt16(start_interval) <= DatumGetInt16(end_interval))
+					valid = false;
+				break;
+			}
+			case INT4OID:
+			{
+				if (DatumGetInt32(start_interval) <= DatumGetInt32(end_interval))
+					valid = false;
+				break;
+			}
+			case INT8OID:
+			{
+				if (DatumGetInt64(start_interval) <= DatumGetInt64(end_interval))
+					valid = false;
+				break;
+			}
+		}
+	}
+	else
+	{
+		Assert(interval_type == INTERVALOID);
+		valid = DatumGetBool(DirectFunctionCall2(interval_gt,
```

Comment:
Interval can be much larger than the valid timestamp/date type on the table, so we should probably check that the interval is not bigger than the valid timestamp range too (e.g., TIMESTAMP_END - TIMESTAMP_MIN).